app/experimenter/experiments/forms.py

```diff
@@ -356,6 +416,45 @@ def clean_population_percent(self):
     def is_valid(self):
         return super().is_valid() and self.variants_formset.is_valid()
 
+    def clean(self):
+        cleaned_data = super().clean()
+
+        # Since we allow '__all__' to be a valid choice for the 'locales'
+        # field, we can't actually use it but instead "transfer" this
+        # to another field which is the 'all_locales'.
+        # Also, the instance.locales field is a many-to-many field
+        # that stores references by primary key so we need to convert
+        # codes to IDs.
+        if "locales" in cleaned_data:
```

Comment:
Same here, if we remove the dependency on exploding it into two fields on the model, we should be able to move this validation into a form field clean method.